pkg/kube/kube_handler.go

```diff
@@ -208,13 +208,9 @@ type ValidationResponse struct {
 	Message string `json:"message"`
 }
 
-// This interface is explicitly private so that it cannot be used in function
-// args, so that call-sites cannot accidentally pass a service handler in place
-// of a user handler.
-// TODO(mnelson): We could instead just create a UserHandler interface which embeds
-// this one and adds one method, to force call-sites to explicitly use a UserHandler
-// or ServiceHandler.
-type handler interface {
+// AuthedHandler exposes Kubernetes resources as either a given user or as a service
+// Note: This should be retrieved as the result of AuthHandler
+type AuthedHandler interface {
```

Comment:
I needed to expose this but I am not sure I understood your comment here: https://github.com/kubeapps/kubeapps/pull/1681#discussion_r412652252

Reply:
Yes, the thing is that I now I am in that position, I want the user of `AuthedHandler` to call it regardless if it's a `UserHandler` or a `ServiceHandler`:

https://github.com/kubeapps/kubeapps/pull/2310/files#diff-7882fe4bc8aaae69a41f78e120b9b973ddec3b087e8eacec396457cb06675f29R274

Note that the decision is made there and the rest is used whatever that resolves. In any case, it's true that it can be risky and misused. Will revert this and try to find a better solution (but it may require to duplicate code).

Reply:
Adding a bit more of color here. There are two things that will be retrieved with the handler: an apprepo and the secrets related to that repo (e.g. CA file or basic auth). For that we can use either the svc handler (for global apprepos) or the user handler (for namespaced apprepos). We would need to duplicate the logic that selects a client depending on the repo namespace (unless I find a better solution but I cannot think of any).